packages/components/nodes/vectorstores/Vectara_Upsert/Vectara_Upsert.ts

```diff
@@ -0,0 +1,132 @@
+import { INode, INodeData, INodeOutputsValue, INodeParams } from '../../../src/Interface'
+import { Embeddings } from 'langchain/embeddings/base'
+import { getBaseClasses } from '../../../src/utils'
+import { VectaraStore, VectaraLibArgs, VectaraFilter } from 'langchain/vectorstores/vectara'
+import { Document } from 'langchain/document'
+import { flatten } from 'lodash'
+
+class VectaraExisting_VectorStores implements INode {
+    label: string
+    name: string
+    description: string
+    type: string
+    icon: string
+    category: string
+    baseClasses: string[]
+    inputs: INodeParams[]
+    outputs: INodeOutputsValue[]
+
+    constructor() {
+        this.label = 'Vectara Upsert Document'
+        this.name = 'vectaraExisting'
+        this.type = 'Vectara'
+        this.icon = 'vectara.png'
+        this.category = 'Vector Stores'
+        this.description = 'Upsert documents to Vectara'
+        this.baseClasses = [this.type, 'VectorStoreRetriever', 'BaseRetriever']
+        this.inputs = [
+            {
+                label: 'Vectara Customer ID',
+                name: 'customerID',
+                type: 'string'
+            },
+            {
+                label: 'Vectara Corpus ID',
+                name: 'corpusID',
+                type: 'string'
+            },
+            {
+                label: 'Vectara API Key',
+                name: 'apiKey',
```

Comment:
Thanks for the PR!

We've recently added a new way to put credentials like API Key, ID etc. Can we seperate these 3 `customerID`, `corpusID` and `apiKey` into another credential file similar to [this](https://github.com/FlowiseAI/Flowise/pull/646/files#diff-4101f7761f09aeeaede04c9ba71bf2ecbf1428218bb68112ec311b1691a9fdfb)?

Reply:
Done!